src/java.base/share/classes/java/util/concurrent/ForkJoinPool.java

```diff
@@ -981,9 +981,7 @@ public final ForkJoinWorkerThread newThread(ForkJoinPool pool) {
             boolean isCommon = (pool.workerNamePrefix == null);
             @SuppressWarnings("removal")
             SecurityManager sm = System.getSecurityManager();
-            if (sm == null)
-                return new ForkJoinWorkerThread(null, pool, true, false);
-            else if (isCommon)
+            if (sm != null && isCommon)
```

Comment:
The check should be `if (isCommon && JLA.allowSecurityManager())` as a SM can be set mid-flight even if it is super weird to do that.

Reply:
I think this is what the backport of [JDK-8342905](https://bugs.openjdk.org/browse/JDK-8342905) will do which Martin mentioned he'll do as a follow-up after this one.